### PR TITLE
BAH-435 Date picker date in appointment list/calendar view

### PR DESCRIPTION
**Pull request by smartyrad**
Date picker date in appointment list/calendar view can be cleared with the delete button

### CODE REVIEW

**Comment by smartyrad** [comment]
@mks-d @djazayeri @shruthidipali Can you please review this PR?
 I am not used to writing tests with Jasmine and hence any help with that will be appreciated :)

**Comment by sumanmaity112** [comment]
Hi @smartyrad, 

Can you squash two commits into one? Also, fix the codeclimate issues

**Comment by smartyrad** [comment]
@sumanmaity112 Sorry for the delay, I have squashed the commits and have also fixed the issues pointed out by codeclimate. I would like to get your help in creating the tests for the same.

File: ui/app/appointments/directives/datePicker.js [hunk 1]
```diff
@@ -22,15 +22,23 @@ angular.module('bahmni.appointments')
                 $scope.viewDate = $scope.viewDate && dateUtil.addDays($scope.viewDate, 1);
             };
 
-            $scope.$watch("viewDate", $scope.onChange($scope.viewDate));
+            $scope.$watch("viewDate", function (viewDate) {
+                if (!viewDate) {
+                    $scope.viewDate = $scope.lastValidDate;
```
Comment (mks-d) [inline ui/app/appointments/directives/datePicker.js]: Is there never a case where `$scope.lastValidDate` is not yet defined here?

Isn't it necessary to at least initialize `$scope.lastValidDate` as today?

File: ui/test/unit/appointments/directives/datePicker.spec.js [hunk 2]
```diff
@@ -23,7 +23,7 @@ describe('DatePicker', function () {
         return element;
     };
 
-    it('should init view date to today if undefined', function () {
+    it('should init view date to the last valid date if undefined', function () {
         scope.viewDate = undefined;
         createElement();
         expect(scope.viewDate).toEqual(moment().startOf('day').toDate());
```
Comment (mks-d) [inline ui/test/unit/appointments/directives/datePicker.spec.js]: Shouldn't this check that `scope.viewDate` equals `scope.lastValidDate`?

**Comment by smartyrad** [comment]
@mks-d How shall I go about fixing the Travis build? I have not updated this PR with the latest code yet but the build failures are exactly the same. The main issue is I'm not yet very sure of how tests run using Jasmine.

**Comment by mks-d** [comment]
@smartyrad what is the minimal set of changes to highlight the Jasmine tests problem (without in fact impacting the app at runtime)? Just changing the binding for `onChange` from `"&"` to `"="`?

**Comment by smartyrad** [comment]
Hi @mks-d , Yes, just changing the binding leads to a failure

**Comment by sumanmaity112** [comment]
Hi @mks-d, @smartyrad 
I think it should be fine.

**Comment by mks-d** [comment]
Thanks @sumanmaity112 .
So @smartyrad you should be able to move forward with this now.

**Comment by smartyrad** [comment]
@mks-d Thanks a ton! I'll get to work right away

**Comment by mks-d** [comment]
@smartyrad you don't necessarily have to close this PR, you can keep submitting incremental commits.

**Comment by smartyrad** [comment]
@mks-d  Actually, I had created a new branch locally. So now instead of creating a pr from my master branch, I have a branch dedicated for this issue. That's the reason a new pr is automatically created and I had to close this one to avoid duplication. I'll be fixing the failures in the other pr and will ping you as soon as I'm done :)

**Pull request closed without merging**